### ISSUE TITLE
magit-gerrit does not load with current gerrit

### ISSUE THREAD

**Issue by jimwise** [terranpro/magit-gerrit#25]
I'm getting the following error (and backtrace) when trying to require the current (from MELPA) magit-gerrit while using the current (from MELPA) magit.  I've verified that this happens either via use-package or when using require directly.

    Error:

    Warning (initialization): An error occurred while loading `/home/sa_jawis/.emacs.d/init.el':

    Symbol's value as variable is void: magit-gerrit-popup

    To ensure normal operation, you should investigate and remove the
    cause of the error in your initialization file.  Start Emacs with
    the `--debug-init' option to view a complete error backtrace.

Stacktrace:

    Debugger entered--Lisp error: (void-variable magit-gerrit-popup)
      byte-code("\301^H\302\303\304\305\306\307&^G\210\310\311\312\313\300$\207" [magit-gerrit-popup magit-define-popup "Popup console for magit gerrit command$
      require(magit-gerrit nil noerror)
      (not (require (quote magit-gerrit) nil (quote noerror)))
      (if (not (require (quote magit-gerrit) nil (quote noerror))) (ignore (message (format "Could not load %s" (quote magit-gerrit)))))
      eval-buffer(#<buffer  *load*> nil "/home/sa_jawis/.emacs.d/init.el" nil t)  ; Reading at buffer position 5433
      load-with-code-conversion("/home/sa_jawis/.emacs.d/init.el" "/home/sa_jawis/.emacs.d/init.el" t t)
      load("/home/sa_jawis/.emacs.d/init" t t)
      #[0 "^H\205\262^@     \306=\203^Q^@\307^H\310Q\202;^@ \311=\204^^^@\307^H\312Q\202;^@\313\307\314\315#\203*^@\316\202;^@\313\307\314\317#\203:^@\320\nB^R$
     command-line()
     normal-top-level()

**Comment by terranpro** [comment]
Hi,

That's kind of strange.  Could you check the output of `magit-version` ? and maybe check the file path where `magit-define-popup` is defined?  I don't use MELPA, but the standard load/require seems to be working...

Also, when I checked MELPA the versions/dates are:
magit  20150823.520 
magit-gerrit 20150805.2242

Are these the same for you?

`magit-define-popup` is the new way of building popups in v2.1.x+
Before that they used something called a key-mode.  The changes made to magit-gerrit to support the new popup stuff happened here: 

https://github.com/terranpro/magit-gerrit/commit/91e6590bfc048e9e8ab343123d16c44afc16b231

Not sure why it would give a void-variable error either.  `magit-define-popup` looks like it's a macro that should create a defun named `magit-gerrit-popup`.

**Comment by jimwise** [comment]
There are actually three packages in melpa -- in recent versions, the magit popup support is in a separate package, magit-popup.  magit-define-popup is indeed a macro, and it is defined in magit-popup.el in the magit-popup package.

I've updated to today's release of magit and magit-popup in MELPA, and the problem persists.  From MELPA's POV, I have installed:

  magit              20150824.713  installed             A Git porcelain inside Emacs
  magit-gerrit       20150805.2242 installed             Magit plugin for Gerrit Code Review
  magit-popup        20150822.732  installed             Define prefix-infix-suffix command combos

magit-version shows:

    Magit 20150824.713, Git 2.4.8, Emacs 24.5.1
